### ISSUE TITLE
integ-runner: integ-runner implictly depends on ts-node

### ISSUE THREAD

**Issue by tmokmss** [aws/aws-cdk#23710]
### Describe the bug

Hi, I found a tiny minor issue regarding integ-runner. It'd be great if you check it when you get a chance.

Starting from 2.60.0, it seems @aws-cdk/integ-runner implicitly depends on ts-node.

https://github.com/aws/aws-cdk/blob/aca8a256dcaf89b53f7af4f308b2f23e2e766902/packages/%40aws-cdk/integ-runner/lib/runner/integration-tests.ts#L251

Without ts-node, the error below happens:

```
$ yarn integ-runner --update-on-failed
yarn run v1.22.17
Verifying integration test snapshots...

node:internal/modules/cjs/loader:988
  throw err;
  ^

Error: Cannot find module 'ts-node/register'
Require stack:
- internal/preload
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:985:15)
    at Function.Module._load (node:internal/modules/cjs/loader:833:27)
    at Module.require (node:internal/modules/cjs/loader:1057:19)
    at Module._preloadModules (node:internal/modules/cjs/loader:1332:12)
    at loadPreloadModules (node:internal/bootstrap/pre_execution:583:5)
    at prepareMainThreadExecution (node:internal/bootstrap/pre_execution:95:3)
    at node:internal/main/run_main_module:9:1 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ 'internal/preload' ]
}
  ERROR      integ.nodejs-build 0.088s
      Command exited with status 1
```

As for usual CDK apps, ts-node is installed by default so it shouldn't be a big problem. In CDK construct library templates generated by projen, however, ts-node is not installed by default so this library behaves somewhat confusing.
Isn't it be more clear if it is explicitly specified in package.json of integ-runner as peerDependency or Dependency?

### Expected Behavior

When installing @aws-cdk/integ-runner, ts-node is installed automatically or at least suggested as a peer dependency.

It will help us a lot if there is more clear way to know the requirements of ts-node, or it is installed automatically.

### Current Behavior

We get an error without ts-node. Also we need to carefully inspect the error and manually install ts-node.

### Reproduction Steps

Use integ-runner for `*integ.ts` files without installing ts-node.

### CDK CLI Version

2.60.0

### Framework Version

2.60.0

### Node.js Version

v16.18.1

### OS

macOS

### Language

Typescript

**Comment by mrgrain** [comment]
Hi @tmokmss 

That's a good question. To me it's in the same realm as running `integ-runner --language=python` which also doesn't come with its own dependencies.

Peer Dependencies these aren't much different from regular dependencies any more, in the sense that they are automatically installed. So adding `ts-node` would be quite the extra load.

Maybe we should check for it's presence and print a nice error?

**Comment by tmokmss** [comment]
Hi @mrgrain, thanks. I agree with that adding ts-node as a dependency is technically not a proper solution, and raising a nicer error is enough for the problem. At the same time, I also tend to think installing ts-node automatically would be more user-friendly and a huge benefit since I believe most users will use integ-runner with TypeScript. Also, integ-runner is usually added as a dev dependency so it shouldn't much affect users who cares CDK package size.

**Comment by mrgrain** [comment]
At some point I had the idea of checking `cdk.json` for the app string. But the parsing seemed very unreliable.

@tmokmss I'm curious - how did you come about finding this out? Did you add integ-runner to a JS only app?
Really just trying to understand the scenario here. I would have thought that anyone running TS integ tests would already have ts-node installed.


PS: Downgrading to p2 since this is an experimental package. But it's on our backlog for integ-runner.

**Comment by tmokmss** [comment]
@mrgrain Sure:) I found this when developing a CDK construct library by using projen ([example](https://github.com/tmokmss/deploy-time-build/blob/main/.projenrc.js)). The [projen template](https://projen.io/api/API.html#projen-awscdk-awscdkconstructlibrary) does not include ts-node by default, so just adding `@aws-cdk/integ-runner` won't work.

**Comment by tmokmss** [comment]
yeah that's why I set an additional build step [here](https://github.com/tmokmss/deploy-time-build/blob/545d7f93ad8eb05efeac4058d20939d107b611cb/.projenrc.js#L27); iirc `yarn build` only builds files in the `src` directory by default so I needed this line (to build .ts files in the `test` directory).

I believe some guideline about using integ-runner for a CDK construct library will also help a lot (maybe after graduation from experimental).

**Comment by mrgrain** [comment]
Absolutely! And native projen support is definitely on my list 😅

**Comment by tmokmss** [comment]
Cool, I'm looking forward to it ❤️ 
I didn't know about `.projenrc.ts` as `npx projen new awscdk-construct` just initializes with a projenrc.js...

**Comment by mrgrain** [comment]
> I didn't know about `.projenrc.ts` as `npx projen new awscdk-construct` just initializes with a projenrc.js...

Yes, I hate this. But haven't found a good way yet to change the default without breaking everyone :(
